### ISSUE TITLE
sed commands don't work on MacOS

### ISSUE THREAD

**Issue by eric-engberg** [artemiscloud/activemq-artemis-operator#373]
**Describe the bug**
On MacOS the -i flag of sed requires a file extension.  Currently the Makefile and the hack script use `sed -i -e 'statement' filename.yaml`
Running this on Mac will not edit the file in place as desired but instead will create a new file named 'filename.yaml-e' as it takes the -e as the value for -i and not the -e flag for sed.

**To Reproduce**
1. Download any current version
3. Attempt to run 'make generate-deploy'

**Platform (please complete the following information):**
 - OS: MacOS
 - Version 1.0.6
 - Go go version go1.19.2 darwin/arm64

**Expected behavior**
The sed commands should edit files in place
On MacOS it would look like the following 
`sed -i "" -e 's/kind: Role/kind: ClusterRole/' "$destdir/cluster_role.yaml"`

I'm not sure if we can just add the "" and it'll work fine on linux and macos both.. otherwise might have to do OS detection and have different commands.  I can test on linux later if nobody else can.

**Comment by davidlanouette** [comment]
This has been tested on both MacOS and Fedora.

**Comment by davidlanouette** [comment]
NOTE: While this fixes the reported problem, I'm wondering why the 'creationTime' field needs to be removed at all?  Was this causing some issue previously?

**Comment by davidlanouette** [comment]
This should have been fixed with PR #530.  It looks like that PR didn't reference this issue correctly.

**Comment by brusdev** [comment]
@davidlanouette you are right, this issue has been fixed by PR #530. Thanks for pointing this out.
